### ISSUE TITLE
Templates working on Windows - but not on Mac

### ISSUE THREAD

**Issue by leastprivilege** [dotnet/templating#1397]
Could you please have a look at this template nuget:

https://www.nuget.org/packages/IdentityServer4.Templates/

source code is here:
https://github.com/IdentityServer/IdentityServer4.Templates

The templates work as expected on Windows, on Mac the command just returns (without the usual "temlate crated" message - and directory is empty, e.g.

`dotnet new is4inmem` 

This is SDK version 2.1.4 on both platforms. This used to work a couple of days ago...

thanks.

**Comment by akamud** [comment]
+1 for me. Same version.

**Comment by seancpeters** [comment]
I've been able to repro this issue using the 2.3.0 version of IdentityServer.Templates, but not with the 2.2.1 version. Since the 2.3.0 version was released at about the same time as this issue was reported, I'm guessing that your problems began after installing the newer version.

The problem is occurring with the 2.3.0 version of the IdentityServer4.Templates because the 2.3.0 nupkg includes some .dll's (but 2.2.1 did not). In general, this is an unsupported scenario. During the installation process, on Windows the templates get successfully installed. But on MacOS / *nix, the install process corrupts the users configuration of dotnet new, making it appear the templates are installed, when in fact they're not (this is a bug).

I've identified why the behavior diverges based on platform, and we are working on a plan for solving this issue, providing safeguards to avoid the Mac/*nix problem, and provide better feedback for template authors during the install process as to what actually happened & went wrong.

For the current release of dotnet new, these templates should work properly if the .dll's can be removed from the template nupkg. One possibility would be to package the .dll's in a separate nupkg on nuget, and setup the template(s) requiring the .dll's to acquire the dll nupkg during the restore process.
